conans/tools.py

```
@@ -392,47 +392,111 @@ def get_solaris_version_name(self, version):
-        elif self._os_info.with_yum:
-            update_command = "%syum check-update" % sudo_str
-        elif self._os_info.is_macos:
-            update_command = "brew update"
-
-        if update_command:
-            print("Running: %s" % update_command)
-            if self._runner(update_command, True) != 0:
-                raise ConanException("Command '%s' failed" % update_command)
+        self._is_up_to_date = True
+        self._tool.update()
 
-    def install(self, package_name):
+    def install(self, packages, update=True, force=False):
         '''
             Get the system package tool install command.
         '''
-        sudo_str = "sudo " if self._sudo else ""
-        install_command = None
-        if self._os_info.with_apt:
-            install_command = "%sapt-get install -y %s" % (sudo_str, package_name)
-        elif self._os_info.with_yum:
-            install_command = "%syum install -y %s" % (sudo_str, package_name)
-        elif self._os_info.is_macos:
-            install_command = "brew install %s" % package_name
-
-        if install_command:
-            print("Running: %s" % install_command)
-            if self._runner(install_command, True) != 0:
-                raise ConanException("Command '%s' failed" % install_command)
-        else:
-            print("Warn: Only available for linux with apt-get or yum or OSx with brew")
-            return None
+        packages = [packages] if isinstance(packages, str) else list(packages)
+        if not force and self._installed(packages):
+            return
+        if update and not self._is_up_to_date:
+            self.update()
+        self._install_any(packages)
+
+    def _installed(self, packages):
+        for pkg in packages:
+            if self._tool.installed(pkg):
+                print("Package already installed: %s" % pkg)
+                return True
+        return False
+
+    def _install_any(self, packages):
+        if len(packages) == 1:
```

I meant `SystemPackageTool.install` will raise different kind of errors if there's one package or a list of packages. The for loop will swallow exceptions if there are multiple packages, and raise a generic exception if none of the packages were installed.
I wanted to keep the methods of each specific package manager as simple as possible, that's why they only accept `str`.